src/gui/creds/webflowcredentials.cpp

```diff
@@ -273,15 +275,35 @@ void WebFlowCredentials::slotReadPasswordJobDone(Job *incomingJob) {
     QKeychain::ReadPasswordJob *job = static_cast<ReadPasswordJob *>(incomingJob);
     QKeychain::Error error = job->error();
 
+    // If we could not find the entry try the old entries
+    if (!_keychainMigration && error == QKeychain::EntryNotFound) {
+        _keychainMigration = true;
+        fetchFromKeychainHelper();
+        return;
+    }
+
     if (error == QKeychain::NoError) {
         _password = job->textData();
         _ready = true;
         _credentialsValid = true;
     } else {
         _ready = false;
     }
-
     emit fetched();
+
+    // If keychain data was read from legacy location, wipe these entries and store new ones
+    if (_keychainMigration && _ready) {
+        persist();
+        deleteOldKeychainEntries();
```

Comment:
is there any need to reset _keychainMigration back to false?

Reply:
Not needed. But I added it anyway.